test/unit/attestation.exs

```diff
@@ -0,0 +1,112 @@
+defmodule Unit.AttestationTest do
+  alias LambdaEthereumConsensus.Beacon.BeaconChain
+  alias LambdaEthereumConsensus.Libp2pPort
+  alias LambdaEthereumConsensus.P2P.Gossip.Attestation
+  alias LambdaEthereumConsensus.P2P.Metadata
+  alias LambdaEthereumConsensus.Store.Db
+  alias Types.AttestationData
+  alias Types.Checkpoint
+
+  use ExUnit.Case
+  use Patch
+
+  doctest Attestation
+
+  setup %{tmp_dir: tmp_dir} do
+    Application.fetch_env!(:lambda_ethereum_consensus, ChainSpec)
+    |> Keyword.put(:config, MinimalConfig)
+    |> then(&Application.put_env(:lambda_ethereum_consensus, ChainSpec, &1))
+
+    start_link_supervised!({Db, dir: tmp_dir})
+    start_link_supervised!(Attestation)
+    patch(Libp2pPort, :subscribe_to_topic, fn _, _ -> :ok end)
+    patch(Libp2pPort, :join_topic, fn -> :ok end)
+    patch(Libp2pPort, :leave_topic, fn -> :ok end)
+    patch(Libp2pPort, :update_enr, fn _, _, _ -> :ok end)
+    patch(Libp2pPort, :validate_message, fn _, _ -> :ok end)
+    patch(BeaconChain, :get_fork_digest, fn -> "9999" end)
+    patch(BeaconChain, :get_fork_version, fn -> "9999" end)
+    patch(Metadata, :get_metadata, fn -> %{attnets: nil, syncnets: nil} end)
```

Comment:
If we're gonna mock everything to only test the db behavior, we might as well move everything related to the db to a separate module (e.g. SubnetInfo) and then test that module. We're not really testing the Attestation module here.